test/ui/signin/sign_in_view_model_test.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:survey_flutter_ic/api/exception/network_exceptions.dart';
+import 'package:survey_flutter_ic/ui/signin/sign_in_view_model.dart';
+import 'package:survey_flutter_ic/ui/signin/sign_in_view_state.dart';
+import 'package:survey_flutter_ic/usecase/base/base_use_case.dart';
+
+import '../../mocks/generate_mocks.mocks.dart';
+
+void main() {
+  group('SignInViewModel', () {
+    late MockSignInUseCase mockSignInUseCase;
+    late ProviderContainer container;
+    late SignInViewModel viewModel;
+
+    setUp(() {
+      mockSignInUseCase = MockSignInUseCase();
+
+      container = ProviderContainer(overrides: [
+        signInViewModelProvider
+            .overrideWith((ref) => SignInViewModel(mockSignInUseCase))
+      ]);
+      addTearDown(() => container.dispose());
+      viewModel = container.read(signInViewModelProvider.notifier);
+    });
+
+    test('When initializing SignInViewModel, its state is Init', () {
```

Comment:
better use a more consolidated writing format i.e:

When ..., it ....

> When initializing SignInViewModel, its state is Init
👉  When initializing SignInViewModel, it returns/has init state

to align with the latter one:
> When calling signIn success, it returns Success state

use `Success` `Fail` `Init` (Pascal case) to indicate the state name. And be consistent everywhere else